src/main/java/com/minecolonies/coremod/event/EventHandler.java

```diff
@@ -210,6 +210,57 @@ public static void onChunkUnLoad(final ChunkEvent.Unload event)
         }
     }
 
+    /**
+     * Called right before dimension change event, used to remove the player from an existing colony
+     *
+     * @param event dim travel event.
+     */
+    @SubscribeEvent(priority = LOWEST)
+    public static void onEntityTravelToDimensionEvent(final EntityTravelToDimensionEvent event)
+    {
+        if (event.getEntity() instanceof ServerPlayerEntity && !event.isCanceled())
+        {
+            final ServerPlayerEntity player = (ServerPlayerEntity) event.getEntity();
+            final Chunk oldChunk = player.world.getChunk(player.chunkCoordX, player.chunkCoordZ);
+            final IColonyTagCapability oldCloseColonies = oldChunk.getCapability(CLOSE_COLONY_CAP, null).orElse(null);
+            // Remove visiting/subscriber from old colony
+            if (oldCloseColonies.getOwningColony() != 0)
```

Comment:
Null check please! Since you are explicitly returning null for the orElse(...) and it is possible that the capability does not exist for some reason or another, a null check is required here.